src/index.ts

```diff
@@ -45,10 +45,17 @@ const indexer = (
   const transform = (documents: SanityDocumentStub[]) => {
     const records: AlgoliaRecord[] = documents.map(
       (document: SanityDocumentStub) => {
-        return Object.assign(standardValues(document), serializer(document))
+        const serializedDocs = serializer(document)
+        if (Array.isArray(serializedDocs)) {
+          return serializedDocs.map((chunk, index) =>
+            Object.assign(standardValues(chunk, String(index)), chunk)
+          )
+        } else {
+          return Object.assign(standardValues(document), serializedDocs)
+        }
```

Comment:
Check if the object returned from the `serializer` is an array... if it is, we know we need to map over each item in that array to add the standard values.